src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -435,7 +435,7 @@ public static ulong GetKey(ReadOnlySpan<byte> propertyName)
         public static Type GetElementType(Type propertyType, Type parentType, MemberInfo memberInfo, JsonSerializerOptions options)
         {
             // We want to handle as the implemented collection type, if applicable.
-            Type implementedType = GetImplementedCollectionType(propertyType);
+            Type implementedType = GetImplementedCollectionType(parentType, propertyType, null, out _, options);
```

Comment:
nit: Use named argument for `null` here/below.

Reply:
Added a commit for this. https://github.com/dotnet/corefx/pull/40435 ports it to master.